lib/tasks/packages.rake

```diff
@@ -97,4 +97,17 @@ namespace :goodcity do
   task :compute_quantities, [:param] => [:environment] do |_, args|
     Goodcity::PackageSetup.compute_quantities(rehearse: args[:param].eql?('rehearse'))
   end
+
+  # rails goodcity:set_offer_for_packages
+  desc "update offers for packages"
+  task set_offer_for_packages: :environment do
+    Item.find_in_batches(batch_size: 100).each do |items|
```

Comment:
I think it would be faster to start with a much smaller list as this could get potentially very large. Something like this? (untested)

```Package.inventorize.received.joins(:item).includes(:item, :offers_package).where('items.offer_id IS NOT NULL').find_each do |package|
  packages.assign_offer unless packages.offers_packages.map(&:offer_id).include(package.item.offer_id)
```

Reply:
Updated. Thanks @steveyken 👍🏻